be/src/olap/rowset/segment_v2/inverted_index_fs_directory.cpp

```diff
@@ -335,6 +350,84 @@ int64_t DorisFSDirectory::FSIndexOutput::length() const {
+            "init",
+            {
+                _CLTHROWA(CL_ERR_IO,
+                          "debug point: test throw error in fsindexoutput init mock error");
+            })
+}
+
+DorisFSDirectory::FSIndexOutputV2::~FSIndexOutputV2() {}
+
+void DorisFSDirectory::FSIndexOutputV2::flushBuffer(const uint8_t* b, const int32_t size) {
+    if (_index_v2_file_writer != nullptr && b != nullptr && size > 0) {
+        Slice data {b, (size_t)size};
+        DBUG_EXECUTE_IF(
+                "DorisFSDirectory::FSIndexOutput._mock_append_data_error_in_fsindexoutput_"
+                "flushBuffer",
+                { return; })
+        Status st = _index_v2_file_writer->append(data);
+        DBUG_EXECUTE_IF(
+                "DorisFSDirectory::FSIndexOutput._status_error_in_fsindexoutput_flushBuffer", {
+                    st = Status::Error<doris::ErrorCode::INVERTED_INDEX_CLUCENE_ERROR>(
+                            "flush buffer mock error");
+                })
+        if (!st.ok()) {
+            LOG(WARNING) << "File IO Write error: " << st.to_string();
+            _CLTHROWA(CL_ERR_IO, "writer append data when flushBuffer error");
+        }
+    } else {
+        if (_index_v2_file_writer == nullptr) {
+            LOG(WARNING) << "File writer is nullptr in DorisFSDirectory::FSIndexOutputV2, "
+                            "ignore flush.";
+        } else if (b == nullptr) {
+            LOG(WARNING) << "buffer is nullptr when flushBuffer in "
+                            "DorisFSDirectory::FSIndexOutput";
+        }
+    }
+}
+
+void DorisFSDirectory::FSIndexOutputV2::close() {
+    try {
+        BufferedIndexOutput::close();
+        DBUG_EXECUTE_IF(
+                "DorisFSDirectory::FSIndexOutput._throw_clucene_error_in_bufferedindexoutput_"
+                "close",
+                {
+                    _CLTHROWA(CL_ERR_IO,
+                              "debug point: test throw error in bufferedindexoutput close");
+                })
+    } catch (CLuceneError& err) {
+        LOG(WARNING) << "FSIndexOutputV2 close, BufferedIndexOutput close error: " << err.what();
+        if (err.number() == CL_ERR_IO) {
+            LOG(WARNING) << "FSIndexOutputV2 close, BufferedIndexOutput close IO error: "
+                         << err.what();
+        }
+        _CLTHROWA(err.number(), err.what());
+    }
+    if (_index_v2_file_writer) {
+        auto ret = _index_v2_file_writer->close();
+        DBUG_EXECUTE_IF("DorisFSDirectory::FSIndexOutput._set_writer_close_status_error",
+                        { ret = Status::Error<INTERNAL_ERROR>("writer close status error"); })
+        if (!ret.ok()) {
+            LOG(WARNING) << "FSIndexOutputV2 close, stream sink file writer close error: "
+                         << ret.to_string();
+            _CLTHROWA(CL_ERR_IO, ret.to_string().c_str());
+        }
+    } else {
+        LOG(WARNING) << "File writer is nullptr, ignore finalize and close.";
```

Comment:
It's a fatal error here, log warining is not enough. You should return throw exception.

Reply:
done